runtime/reconcilers/connector.go

```diff
@@ -59,6 +60,35 @@ func (r *ConnectorReconciler) Reconcile(ctx context.Context, n *runtimev1.Resour
 		return runtime.ReconcileResult{Err: errors.New("not a connector")}
 	}
 
+	// Get and sync repo
+	repo, release, err := r.C.Runtime.Repo(ctx, r.C.InstanceID)
+	if err != nil {
+		return runtime.ReconcileResult{Err: fmt.Errorf("failed to access repo: %w", err)}
+	}
+	defer release()
+	err = repo.Sync(ctx)
+	if err != nil {
+		return runtime.ReconcileResult{Err: fmt.Errorf("failed to sync repo: %w", err)}
+	}
+
+	// Get instance
+	inst, err := r.C.Runtime.Instance(ctx, r.C.InstanceID)
+	if err != nil {
+		return runtime.ReconcileResult{Err: fmt.Errorf("failed to find instance: %w", err)}
+	}
+
+	// Parse the project
+	parser, err := rillv1.Parse(ctx, repo, r.C.InstanceID, inst.Environment, inst.OLAPConnector)
+	if err != nil {
+		return runtime.ReconcileResult{Err: fmt.Errorf("failed to parse: %w", err)}
+	}
+
+	// Update instance connectors
+	err = r.C.Runtime.UpdateInstanceWithRillYAML(ctx, inst.ID, parser, false)
+	if err != nil {
+		return runtime.ReconcileResult{Err: fmt.Errorf("failed to update instance: %w", err)}
+	}
```

Comment:
The file watcher already parsed the file and created the `Connector` resource (returned from `self.GetConnector()` above). Doing a repo sync (clone or pull) and full project parse again here is redundant, and potentially quite slow if there are many connectors. 

Instead, you will probably want to add new functions, separate from `UpdateInstanceWithRillYAML`, for upserting/deleting connectors for an instance. They should also take care only to update in the DB (invalidating various caches) if the connector actually changed. (Since the reconciler can trigger even if there are no changes, e.g. on process restart.)

The delete case needs to be handled in the `if self.Meta.DeletedOn != nil {` case below.